sql/catalyst/src/main/java/org/apache/spark/sql/connector/expressions/filter/Predicate.java

```diff
@@ -146,4 +149,210 @@ public class Predicate extends GeneralScalarExpression {
   public Predicate(String name, Expression[] children) {
     super(name, children);
   }
+
+  public org.apache.spark.sql.sources.Filter toV1() {
```

Comment:
We shouldn't add this public API. Can we have a private internal util function to do it?